packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -4129,6 +4129,15 @@ namespace Tpetra {
                                x_lcl_1d, false, false);
     }
     else {
+      // FENCE REVIEW - NOT TESTED
+      //   Testing: This code is NOT exercised by unit tests.
+      //   GTX960:  N/A
+      //   White:   N/A
+      //   Plan:    Potentially add unit testing for this code.
+      //            Potentially remove this fence but following code appears
+      //            to need refactoring first.
+      //   Notes:
+
```

Comment:
@kddevin This fence is necessary. rowValues are from local crs matrix (kokkos crs matrix) with device type where the device currently has UVM. Without UVM, this routine is not correct as line 4150 access the values from the host code. 

Again, if we can define the state of the local object, we can skip the fence.

Reply:
Removed with PR 7560